File: easy/14. Nth Fibonacci/nth_fibonacci.go
package main

func GetNthFib(n int) int {
  fibies := make([]int, n+1)
  fibies[0] = 0
  fibies[1] = 1
  for i := 2; i <= n; i++ {
    fibies[i] = fibies[i-1] + fibies[i-2]
  }
  return fibies[n-1]
}

File: dsa/easy/Nth Fibonacci/nth_fibonacci.cpp
#include <map>

int getNthFib(int n) {
  std::map<int, int> fibs;
  fibs[0] = 0;
  fibs[1] = 1;
  for (int i = 2; i <= n; i++) {
    fibs[i] = fibs[i - 1] + fibs[i - 2];
  }
  return fibs[n-1];
}
